src/Core/Silk.NET.SilkTouch/StringMarshaller.cs

```diff
@@ -0,0 +1,242 @@
+                [UnmanagedType.LPWStr] = MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, _sysMarshal, IdentifierName("StringToHGlobalUni")),
+                [UnmanagedType.LPStr] = MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, _sysMarshal, IdentifierName("StringToHGlobalAnsi")),
+                [UnmanagedType.LPTStr] = MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, _sysMarshal, IdentifierName("StringToHGlobalAuto")),
+            };
+        private static Dictionary<UnmanagedType, ExpressionSyntax> _stringFromPtr =
+            new Dictionary<UnmanagedType, ExpressionSyntax>
+            {
+                [UnmanagedType.BStr] = MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, _sysMarshal, IdentifierName("PtrToStringBSTR")),
+                [UnmanagedType.LPWStr] = MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, _sysMarshal, IdentifierName("PtrToStringUni")),
+                [UnmanagedType.LPStr] = MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, _sysMarshal, IdentifierName("PtrToStringAnsi")),
+                [UnmanagedType.LPTStr] = MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, _sysMarshal, IdentifierName("PtrToStringAuto")),
+            };
+        private static Dictionary<UnmanagedType, ExpressionSyntax> _freeString =
+            new Dictionary<UnmanagedType, ExpressionSyntax>
+            {
+                [UnmanagedType.BStr] = MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, _sysMarshal, IdentifierName("FreeBSTR")),
+                [UnmanagedType.LPWStr] = _freeHGlobal,
+                [UnmanagedType.LPStr] = _freeHGlobal,
+                [UnmanagedType.LPTStr] = _freeHGlobal,
+            };
+
+        private static Dictionary<UnmanagedType, ExpressionSyntax> _allocString =
+            new Dictionary<UnmanagedType, ExpressionSyntax>
+            {
+                [UnmanagedType.BStr] =
+                    // Silk.NET.Core.AllocBStr
+                    MemberAccessExpression
+                    (
+                        SyntaxKind.SimpleMemberAccessExpression,
+                        MemberAccessExpression
+                        (
+                            SyntaxKind.SimpleMemberAccessExpression,
+                            MemberAccessExpression
+                            (
+                                SyntaxKind.SimpleMemberAccessExpression,
+                                MemberAccessExpression
+                                (
+                                    SyntaxKind.SimpleMemberAccessExpression, IdentifierName("Silk"),
+                                    IdentifierName("NET")
+                                ), IdentifierName("Core")
+                            ), IdentifierName("Native")
+                        ), IdentifierName("AllocBStr")
```

Comment:
By the looks of things this will generate:
```
Silk.NET.Core.Native.AllocBStr(blah)
```
which isn't correct. Another identifier needs to be added such that it generates:
```
Silk.NET.Core.Native.SilkMarshal.AllocBStr
```